joern-cli/frontends/csharpsrc2cpg/src/test/scala/io/joern/csharpsrc2cpg/querying/ast/MemberTests.scala

```diff
@@ -178,13 +178,16 @@ class MemberTests extends CSharpCode2CpgFixture {
         |  string color;                // field
         |  static int maxSpeed = 200;   // field
         |  static int nonInitMaxSpeed;  // field
+        |  int b;
         |
         |  static Car() { // static constructor
         |     this.nonInitMaxSpeed = 2000;
         |  }
         |
         |  public Car() {
         |     this.color = "red";
+        |     b = 1000;
```

Comment:
Could you add a test case where there is a local variable shadowing the field, so that it correctly gets determined as a non-field access? e.g.
```java
public class Foo {
   int a;
   public Foo() {
      var a = 2;
      this.a = a;
   }
}
```